kuka_eki_hw_interface/krl/README.md

```diff
@@ -0,0 +1,94 @@
+
+* **Joint names** corresponding to the robot description (URDF or .xacro).
+* **IP addresses** and **port numbers** corresponding to the EKI setup specified via the above XML files.
+
+We recommend that you copy the configuration files, edit the copies for your needs and use these files to create your own launch file. A template will be provided at a later time. However, at this time, have a look at `test_hardware_interface.launch`, `test_params.yaml`, `controller_joint_names.yaml` and `hardware_controllers.yaml` to achieve a working test-launch.
+
+In order to successfully launch the **kuka_eki_hw_interface** a parameter `robot_description` needs to be present on the ROS parameter server. This parameter can be set manually or by adding this line inside the launch file (replace support package and .xacro to match your application):
+
+```
+<param name="robot_description" command="$(find xacro)/xacro.py '$(find kuka_kr6_support)/urdf/kr6r900sixx.xacro'"/>
+```
+
+Make sure that the line is added before the `kuka_eki_hw_interface` itself is loaded.
+
+## 4. Testing
+At this point you are ready to test the EKI interface. Before the test, make sure that:
+
+* You have specified the `eki/state_server_address`, `eki/state_server_port`, `eki/command_server_address` and `eki/command_server_port` of the **kuka_eki_hw_interface** to correspond with the KRL files on the controller.
+* You have a launch-file loading the network parameters, robot description, kuka_hw_interface, hardware controller and controller joint names.
+
+The next steps describe how to launch the test file:
+
+* On the robot, start the servers
+
+1. On the teach pad, enter mode **T1** for testing purposes.
+2. Navigate to `KRC:\R1\Program` and select `kuka_eki_hw_interface.src`.
+3. Press and hold an enabling switch and the run/play-button. The robot will release its breaks and a message like **Programmed path reached (BCO)** will be shown at the teach pad.
+4. Press and hold again. The robot is now waiting for clients to connect.
+
+* On your ROS PC, in a new terminal:
+
+```
+$ roscore
+```
+
+* Open a new terminal and roslaunch the previously configured launch-file:
+
+```
+$ roslaunch kuka_eki_hw_interface test_hardware_interface.launch
+```
+
+The **kuka_eki_hw_interface** is now connected and streaming commands.
+
+6. In a new terminal:
+
+```
+$ rosrun rqt_joint_trajectory_controller rqt_joint_trajectory_controller
+```
+
+Choose **controller manager ns** and **controller** and you should be able to move each robot joint.
+
+* Note that T1-mode limits the robot movement velocity and is intended for testing purposes.
+
```

Comment:
Yes, as soon as you send anything to the state server it just starts periodic transmission of the state (irregardless of any joint commands).  It is still possible (but not necessary) to command it.  To disable commanding would require commenting the command server line(s) in the KRL to be commented out or removed.  Is such segmented functionality desirable?

Reply:
>  Is such segmented functionality desirable?

I'm not sure. I was just wondering. I could imagine an application where only state reporting is required (such as using regular hard-coded KRL to control motion, but for syncing sensor recordings still broadcasting joint poses).

Disabling the motion control side might then be a nice thing to be able to do, to avoid accidentally activating it.

Reply:
I am planning on combining the state server and joint commanding to a single interface as discussed.  It this case, disabling the joint command functionality would be achieved by commenting out or deleting the infinite `PTP` loop in the KRL .src file (would most likely be replaced by user KRL code in this scenario anyway).  Do you think good readme documentation of this possibility is sufficient?